### PR TITLE
feat(pie-docs): DSW-1415 add link component shortcode

### PR DESCRIPTION
**Pull request by LTurns**
## Describe your changes (can list changeset entries if preferable)

[Added] - link shortcode component

Screenshot 2023-10-29 at 16 22 00

Not active on any pages yet but needed for the style template.

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: apps/pie-docs/src/__tests__/_11ty/shortcodes/link.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    it('should return the expected HTML for the link', () => {
+        // act
+        const result = createLink(link);
+
+        // assert
+        expect(result).toMatchSnapshot();
+    });
+
+    xit.each([{
```
Comment (FayeCarter) [inline apps/pie-docs/src/__tests__/_11ty/shortcodes/link.test.js:14]: is this meant to be skipped?

**Review by raoufswe — DISMISSED** [review]
looks good to me.

File: apps/pie-docs/src/__tests__/_11ty/shortcodes/link.test.js (hunk 1 above)
Comment (raoufswe) [reply apps/pie-docs/src/__tests__/_11ty/shortcodes/link.test.js:14]: +1

Comment (LTurns) [reply apps/pie-docs/src/__tests__/_11ty/shortcodes/link.test.js]: fixed, thank you!

**Review by FayeCarter — APPROVED** [review]

**Review by raoufswe — APPROVED** [review]

**Pull request merged**